# Lab 18 Exercise 1

## Generic

1.สร้าง console application project

```cmd
dotnet new console --name Lab18_Ex01
```

2.เปลี่ยน code ให้เป็นดังต่อไปนี้

```csharp
var IntCal  = new Calculator<int>();
var IntResult = IntCal.Add(2,3);
System.Console.WriteLine(IntResult);
System.Console.WriteLine("Type of IntCal is "+ IntCal.GetType());

var FloatCal  = new Calculator<float>();
var FloatResult = FloatCal.Add(2.5f, 3f);
System.Console.WriteLine(FloatResult);
System.Console.WriteLine("Type of IntCal is "+ FloatCal.GetType());

class Calculator<T>
{
    public T Add(T a, T b)
    {
        dynamic? number1 = a;
        dynamic? number2 = b;
        return number1 + number2;
    }
}
```

3.Build project โดยการใช้คำสั่ง

```cmd
dotnet build  Lab18_Ex01
```

ถ้ามีที่ผิดพลาดในโปรแกรม ให้แก้ไขให้ถูกต้อง

4.บันทึกผลที่ได้จากการรันคำสั่งในข้อ 3

Restore complete (0.3s)
  Lab18_Ex01 succeeded (0.4s) → Lab18_Ex01\bin\Debug\net9.0\Lab18_Ex01.dll

Build succeeded in 1.2s

5.Run project โดยการใช้คำสั่ง

```cmd
dotnet run --project Lab18_Ex01
```

In [1]:
var IntCal  = new Calculator<int>();
var IntResult = IntCal.Add(2,3);
System.Console.WriteLine(IntResult);
System.Console.WriteLine("Type of IntCal is "+ IntCal.GetType());

var FloatCal  = new Calculator<float>();
var FloatResult = FloatCal.Add(2.5f, 3f);
System.Console.WriteLine(FloatResult);
System.Console.WriteLine("Type of IntCal is "+ FloatCal.GetType());

class Calculator<T>
{
    public T Add(T a, T b)
    {
        dynamic? number1 = a;
        dynamic? number2 = b;
        return number1 + number2;
    }
}

5
Type of IntCal is Submission#1+Calculator`1[System.Int32]
5.5
Type of IntCal is Submission#1+Calculator`1[System.Single]


6.บันทึกผลที่ได้จากการรันคำสั่งในข้อ 5

5
Type of IntCal is Submission#1+Calculator`1[System.Int32]
5.5
Type of IntCal is Submission#1+Calculator`1[System.Single]

7.อธิบายสิ่งที่พบในการทดลอง